src/main/java/com/exadel/backendservice/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.exadel.backendservice.services.impl;
+
+import com.exadel.backendservice.dto.RoleDto;
+import com.exadel.backendservice.dto.UserDtoWithId;
+import com.exadel.backendservice.entity.RoleEntity;
+import com.exadel.backendservice.entity.UserEntity;
+import com.exadel.backendservice.model.RegistrationRequest;
+import com.exadel.backendservice.repository.RoleEntityRepository;
+import com.exadel.backendservice.repository.UserEntityRepository;
+import com.exadel.backendservice.services.UserService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserServiceImpl.class);
+
+    private final UserEntityRepository userEntityRepository;
+    private final RoleEntityRepository roleEntityRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserServiceImpl(UserEntityRepository userEntityRepository, RoleEntityRepository roleEntityRepository, PasswordEncoder passwordEncoder) {
+        this.userEntityRepository = userEntityRepository;
+        this.roleEntityRepository = roleEntityRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public Boolean saveUser(RegistrationRequest registrationRequest) {
```

Comment:
need to be @Transactional 
as most of service's public methods

Reply:
corrected